src/screens/Customers/CustomerProfile.tsx

```diff
@@ -36,92 +37,137 @@ const styles = (theme: Theme) =>
   });
 
 interface CustomerProps extends RouteComponentProps {
+  match: { params: string; isExact: boolean; path: string; url: string; }
   classes: { root: string; content: string; header: string; button: string; buttonSecondary: string };
+  siteName: string;
   location: any;
 }
 
-function CustomerProfile(props: CustomerProps) {
-  const { classes, match } = props;
-  const customer: CustomerRecord = props.location.state.customer;
+class CustomerProfile extends React.Component<CustomerProps, {}> {
+  render() {
+    const { classes, match } = this.props;
+    const customer = this.props.location.state.customer;
+    const meterReadings = customer.meterReadings;
 
-  const getPaymentButtons = () => {
-    return (
-      <>
-        <Button
-          variant="contained"
-          className={classes.button}
-          color="primary"
-          startIcon={<AddIcon />}
-          disableElevation={true}
-        >
-          Add Payment
-        </Button>
-        <div className={classes.buttonSecondary}>
+    // helper functions for getting data
+    const isCurrentReading = (mr: MeterReadingRecord): boolean => {
+      const period: number = parseInt(mr.date.slice(5, 7));
+      const currPer: number = moment().month() + 1;
+      return period == currPer;
+    };
+
+    const getCurrentReading = (customer: CustomerRecord): MeterReadingRecord | undefined => {
+      return customer.meterReadings.find(function (mr: MeterReadingRecord) {
+        return isCurrentReading(mr);
+      });
+    };
+
+    const isStartingMeter = (mr: MeterReadingRecord): boolean => {
+      const period: number = parseInt(mr.date.slice(5, 7)) % 12;
+      const lastPer: number = moment().month();
+      return period == lastPer;
+    };
+
+    const getStartingMeter = (customer: CustomerRecord): MeterReadingRecord | undefined => {
+      return customer.meterReadings.find(function (mr: MeterReadingRecord) {
+        return isStartingMeter(mr);
+      });
+    };
+
+    const getAmountBilled = (customer: CustomerRecord, currReading: number, startingMeter = 0) => {
+      const tariffPlan = customer.tariffPlans[0];
+      const periodUsage = currReading - startingMeter;
```

Comment:
You should probably add a check on whether or not the current reading is valid or not (if there is no current reading, there should also be no period usage -- this also prevents it from displaying negative numbers when no reading exists)